src/zjs_fs.c

```diff
@@ -413,17 +421,21 @@ static jerry_value_t zjs_read(const jerry_value_t function_obj,
         return zjs_error("offset + length overflows buffer");
     }
 
-    if (handle->mode == MODE_A_PLUS) {
+    // a+ or read from current position (position == null)
+    if ((handle->mode == MODE_A_PLUS) || from_cur) {
         // mode is a+, seek to read position
         if (fs_seek(&handle->fp, handle->rpos, SEEK_SET) != 0) {
             return zjs_error("error seeking to position");
         }
     }
-    // if a position was specified, seek to it before reading
-    if (fs_seek(&handle->fp, (uint32_t)position, SEEK_CUR) != 0) {
-        return zjs_error("error seeking to position");
+    if (!from_cur) {
```

Comment:
Oh and here you want AND NOT A_PLUS or else you're still going to overwrite your seek position you set above. :)